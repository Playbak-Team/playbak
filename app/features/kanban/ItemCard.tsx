/* eslint-disable react/jsx-props-no-spreading */
import React, { useState, useEffect } from 'react';
import { makeStyles, createStyles } from '@material-ui/core/styles';
import { Draggable } from 'react-beautiful-dnd';
import type {
  DroppableProvided,
  DroppableStateSnapshot,
} from 'react-beautiful-dnd';
import EditIcon from '@material-ui/icons/Edit';
import ScheduleIcon from '@material-ui/icons/Schedule';
import HighlightOffIcon from '@material-ui/icons/HighlightOff';
import IconButton from '@material-ui/core/IconButton';
import { EntryCardProps } from '../../types';

const useStyles = makeStyles(() =>
  createStyles({
    entry: {
      backgroundColor: 'white',
      maxWidth: '85%',
      minHeight: '50px',
      maxHeight: 'max-content',
      margin: '10px',
      padding: '10px',
      borderRadius: '3px',
      boxShadow: '2px 2px 1px #ccc',
    },
    entryInner: {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'space-between',
    },
    cardEntryTitle: {
      color: 'black',
      margin: '0',
    },
  })
);

const getItemStyle = (isDragging, draggableStyle) => ({
  // some basic styles to make the items look a bit nicer
  userSelect: 'none',

  // change background colour if dragging
  background: isDragging ? '#97CBFF' : 'white',

  border: isDragging ? '2px solid black' : 'none',

  // styles we need to apply on draggables
  ...draggableStyle,
});

const ItemCard = (props: EntryCardProps) => {
  const { og, id, title, duedate, index, openCardInfo, removeE } = props;
  const classes = useStyles();
  const [urgent, setUrgent] = useState(false);

  useEffect(() => {
    const today = new Date();
    const due = new Date(duedate);

    if ((due.getTime() - today.getTime()) / 3600000 <= 24) {
      setUrgent(true);
    } else {
      setUrgent(false);
    }
  }, [duedate]);

  return (
    <div>
      <Draggable key={id} draggableId={id.toString()} index={index}>
        {(provided: DroppableProvided, snapshot: DroppableStateSnapshot) => (
          <div
            ref={provided.innerRef}
            {...provided.draggableProps}
            {...provided.dragHandleProps}
            className={classes.entry}
            style={getItemStyle(
              snapshot.isDragging,
              provided.draggableProps.style
            )}
          >
            <div className={classes.entryInner}>
              <div
                style={{
                  display: 'flex',
                  flexDirection: 'row',
                  justifyContent: 'space-between',
                }}
              >
                <h3 className={classes.cardEntryTitle}>{title}</h3>
                <div>
                  <IconButton
                    onClick={() => openCardInfo(og)}
                    style={{ marginTop: '-10px', marginRight: '-10px' }}
                  >
                    <EditIcon color="secondary" />
                  </IconButton>
                  <IconButton
                    onClick={() => removeE(og)}
                    style={{ marginTop: '-10px', marginRight: '-10px' }}
                  >
                    <HighlightOffIcon />
                  </IconButton>
                </div>
              </div>
              <div
                style={{
                  display: 'flex',
                  flexDirection: 'row',
                  alignItems: 'center',
                  marginTop: '10px',
                  backgroundColor: urgent ? '#FF8484' : 'transparent',
                  maxWidth: 'max-content',
                  padding: '3px',
                  borderRadius: '3px',
                }}
              >
                <ScheduleIcon style={{ marginRight: '5px' }} />
                {duedate}
              </div>
            </div>
          </div>
        )}
      </Draggable>
    </div>
  );
};

export default ItemCard;
