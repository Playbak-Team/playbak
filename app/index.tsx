import React, { Fragment } from 'react';
import { render } from 'react-dom';
import { AppContainer as ReactHotAppContainer } from 'react-hot-loader';
import { Titlebar, Color } from 'custom-electron-titlebar';
import { remote } from 'electron';
import { history, configuredStore } from './store';
import './app.global.css';

const { Menu } = remote;

const store = configuredStore();

const AppContainer = process.env.PLAIN_HMR ? Fragment : ReactHotAppContainer;

const templateDefault = [
  {
    label: 'Home',
    click: () => history.push('/home'),
  },
  {
    label: 'Apps',
    submenu: [
      {
        label: 'Watch',
        click: () => history.push('/video'),
      },
      {
        label: 'Kanban',
        click: () => history.push('/kanban'),
      },
      {
        label: 'Profile',
        click: () => history.push('/profile'),
      },
    ],
  },
];

document.addEventListener('DOMContentLoaded', () => {
  // eslint-disable-next-line no-new
  new Titlebar({
    backgroundColor: Color.fromHex('#1B1E2B'),
    unfocusEffect: true,
    menu: Menu.buildFromTemplate(templateDefault),
    titleHorizontalAlignment: 'left',
    overflow: 'hidden',
    icon: '../resources/icon.png',
  });
  // eslint-disable-next-line global-require
  const Root = require('./containers/Root').default;
  render(
    <AppContainer>
      <Root store={store} history={history} />
    </AppContainer>,
    document.getElementById('root')
  );
});
