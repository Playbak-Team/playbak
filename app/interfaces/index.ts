interface VideoData {
  name: string;
  videoPath: string;
  pbsPath: string;
  watched: boolean;
}

const emptyVideoData = (): VideoData => ({
  name: '',
  videoPath: '',
  pbsPath: '',
  watched: false,
});

interface Settings {
  name: string;
  LST: string;
  LL: string;
  courses: string[];
  AWKS: string[];
}

const emptySettings = (): Settings => ({
  name: '',
  LST: '',
  LL: '',
  courses: [],
  AWKS: [],
});

interface PBSData {
  name: string;
  chunkSize: number;
  speeds: number[];
}

const emptyPBSData = (): PBSData => ({
  name: '',
  chunkSize: -1,
  speeds: [],
});

interface ProfileStateInterface {
  name: string;
  selectedWorkspace: string;
  availableWorkspaces: string[];
  courses: string[];
  links: string[];
}

interface KanbanStateInterface {
  columns: string[];
  entries: string[];
}

interface Columns {
  name: string;
}

interface Events {
  id: number;
  title: string;
  subtitle: string;
  description: string;
  label: string;
  duedate: string;
  belongsto: string;
}

enum SnackbarSeverity {
  success = 'success',
  info = 'info',
  warning = 'warning',
  error = 'error',
}

interface SnackbarStateInterface {
  snackbarActive: boolean;
  snackbarMessage: string;
  snackbarSeverity: 'success' | 'info' | 'warning' | 'error' | undefined;
}

// eslint-disable-next-line import/prefer-default-export
export {
  ProfileStateInterface,
  KanbanStateInterface,
  VideoData,
  emptyVideoData,
  Settings,
  emptySettings,
  Columns,
  Events,
  PBSData,
  emptyPBSData,
  SnackbarSeverity,
  SnackbarStateInterface,
};
