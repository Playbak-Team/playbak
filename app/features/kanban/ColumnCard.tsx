/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { makeStyles, createStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import { Droppable, Draggable } from 'react-beautiful-dnd';
import type {
  DroppableProvided,
  DroppableStateSnapshot,
} from 'react-beautiful-dnd';
import AddIcon from '@material-ui/icons/Add';
import DeleteForeverIcon from '@material-ui/icons/DeleteForever';
import IconButton from '@material-ui/core/IconButton';
import { ColumnDivProps } from '../../types';
import ItemCard from './ItemCard';

const useStyles = makeStyles(() =>
  createStyles({
    col: {
      display: 'flex',
      flexDirection: 'column',
      maxHeight: '60vh',
      backgroundColor: 'rgb(235,236,240)',
      margin: '20px',
      borderRadius: '5px',
      minWidth: '15vw',
    },
    coltitle: {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'space-between',
      paddingLeft: '12px',
      borderRadius: '5px 5px 5px 5px',
      color: '#171D20',
      maxHeight: 'max-content',
    },
    containerparent: {
      minWidth: '87%',
      minHeight: '60vh',
      maxHeight: '60vh',
      overflow: 'hidden',
      position: 'relative',
      padding: '10px',
      backgroundColor: 'rgb(235,236,240)',
      borderRadius: '0px 0px 5px 5px',
    },
    container: {
      overflowY: 'scroll',
      backgroundColor: 'rgb(235,236,240)',
      position: 'absolute',
      top: '0px',
      bottom: '0px',
      left: '0px',
      right: '-17px',
      color: '#171D20',
      borderRadius: '3px',
    },
    entriescontainer: {
      minWidth: '87%',
      display: 'flex',
      flexDirection: 'column',
      borderRadius: '0px 0px 5px 5px',
    },
  })
);

const getListStyle = (isDraggingOver) => ({
  background: isDraggingOver ? 'lightblue' : 'rgb(235,236,240)',
});

const getColStyle = (isDragging, draggableStyle) => ({
  userSelect: 'none',

  backgroundColor: isDragging ? 'green' : '#FFCF99',

  // styles we need to apply on draggables
  ...draggableStyle,
});

const ColumnCard = (props: ColumnDivProps) => {
  const { name, entries, addE, removeC, index, openCardInfo } = props;
  const classes = useStyles();

  return (
    <Draggable draggableId={`${name}`} index={index}>
      {(p2, s2) => (
        <div
          ref={p2.innerRef}
          {...p2.draggableProps}
          {...p2.dragHandleProps}
          className={classes.col}
          style={getColStyle(s2.isDraggingOver, p2.draggableProps.style)}
        >
          <div className={classes.coltitle}>
            <h3>{name}</h3>
            <IconButton onClick={() => removeC(name)}>
              <DeleteForeverIcon />
            </IconButton>
          </div>
          <Droppable key={name} droppableId={`${name}`} type="ENTRY">
            {(
              provided: DroppableProvided,
              snapshot: DroppableStateSnapshot
            ) => (
              <div className={classes.containerparent}>
                <div
                  className={classes.container}
                  style={getListStyle(snapshot.isDraggingOver)}
                  ref={provided.innerRef}
                  {...provided.droppableProps}
                >
                  <div className={classes.entriescontainer}>
                    {entries !== undefined ? (
                      entries
                        .filter((ob) => ob.split(',')[6] === name)
                        .map((ob, i2) => (
                          <div key={ob.split(',')[0]}>
                            <ItemCard
                              og={ob}
                              id={parseInt(ob.split(',')[0], 10)}
                              title={ob.split(',')[1]}
                              subtitle={ob.split(',')[2]}
                              description={ob.split(',')[3]}
                              label={ob.split(',')[4]}
                              duedate={ob.split(',')[5]}
                              belongsto={ob.split(',')[6]}
                              index={i2}
                              openCardInfo={openCardInfo}
                            />
                          </div>
                        ))
                    ) : (
                      <div />
                    )}
                  </div>
                  {provided.placeholder}
                </div>
              </div>
            )}
          </Droppable>
          <Button
            variant="contained"
            color="primary"
            style={{
              margin: '10px',
              minWidth: '90%',
              display: 'flex',
              flexDirection: 'row',
              justifyContent: 'flex-start',
            }}
            onClick={() => addE(name)}
          >
            <AddIcon style={{ marginRight: '5px' }} />
            Add another card
          </Button>
        </div>
      )}
    </Draggable>
  );
};

export default ColumnCard;
