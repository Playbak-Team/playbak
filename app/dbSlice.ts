import { createSlice, PayloadAction } from '@reduxjs/toolkit';
// eslint-disable-next-line import/no-cycle
import { RootState } from './store';

import { UserInfoInterace } from './interfaces';

import udata from '../db/stores/settings.json';

const writeJsonFile = require('write-json-file');

const initialState: UserInfoInterace = udata;

const dbSlice = createSlice({
  name: 'dbinfo',
  initialState,
  reducers: {
    setName: (state, action: PayloadAction<string>) => {
      state.name = action.payload;
      udata.name = action.payload;
      writeJsonFile('./db/stores/newsettings.json', udata);
    },
    setLST: (state, action: PayloadAction<string>) => {
      state.LST = action.payload;
    },
  },
});

export const { setName, setLST } = dbSlice.actions;

export default dbSlice.reducer;

export const getName = (state: RootState) => state.dbinfo.name;

export const getLST = (state: RootState) => state.dbinfo.LST;
