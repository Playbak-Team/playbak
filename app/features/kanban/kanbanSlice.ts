import { createSlice, PayloadAction } from '@reduxjs/toolkit';
// eslint-disable-next-line import/no-cycle
// import { ipcRenderer } from 'electron';
// eslint-disable-next-line import/no-cycle
import { RootState } from '../../store';

import { KanbanStateInterface } from '../../interfaces';

const initialState: KanbanStateInterface = {
  columns: [],
  entries: [],
};

const kanbanSlice = createSlice({
  name: 'kanban',
  initialState,
  reducers: {
    addColumn: (state, action: PayloadAction<string>) => {
      state.columns.push(action.payload);
      // db call here
    },
    setColumns: (state, action: PayloadAction<string[]>) => {
      state.columns = action.payload;
      // db call here
    },
    removeColumn: (state, action: PayloadAction<string>) => {
      if (state.columns.includes(action.payload)) {
        state.columns.splice(state.columns.indexOf(action.payload), 1);
        // db call here
      }
      state.entries = state.entries.filter(
        (e) => e.split(',')[6] !== action.payload
      );
    },
    addEntry: (state, action: PayloadAction<string>) => {
      state.entries.push(action.payload);
    },
    setEntries: (state, action: PayloadAction<string[]>) => {
      state.entries = action.payload;
    },
  },
});

export const {
  setColumns,
  addColumn,
  removeColumn,
  addEntry,
  setEntries,
} = kanbanSlice.actions;

export default kanbanSlice.reducer;

export const getColumnNames = (state: RootState) => state.kanban.columns;

export const getEntries = (state: RootState) => state.kanban.entries;
