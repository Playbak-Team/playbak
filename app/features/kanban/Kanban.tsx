/* eslint-disable react/jsx-props-no-spreading */
import React, { useState, useEffect } from 'react';
import { makeStyles, createStyles } from '@material-ui/core/styles';
import { useSelector, useDispatch } from 'react-redux';
import Button from '@material-ui/core/Button';
import { DragDropContext, Droppable } from 'react-beautiful-dnd';
import AddCircleIcon from '@material-ui/icons/AddCircle';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import { DragProps } from '../../types';
import {
  getColumnNames,
  getEntries,
  addEntry,
  addColumn,
  removeColumn,
  setColumns,
  replaceEntry,
  setEntries,
} from './kanbanSlice';
import EditEntry from './EditEntry';
import ColumnCard from './ColumnCard';

const useStyles = makeStyles(() =>
  createStyles({
    root: {
      width: 'max-content',
      minWidth: '100%',
      height: '100vh',
      backgroundColor: 'rgb(38, 132, 255)',
      display: 'flex',
      flexDirection: 'row',
    },
    page: {
      display: 'flex',
      flexDirection: 'column',
      backgroundColor: 'rgb(38, 132, 255)',
      width: 'max-content',
      minWidth: '100%',
    },
  })
);

const reorder = (list, startIndex, endIndex) => {
  const result = Array.from(list);
  const [removed] = result.splice(startIndex, 1);
  result.splice(endIndex, 0, removed);

  return result;
};

const move = (
  sourceString,
  destinationString,
  source,
  destination,
  droppableSource,
  droppableDestination
) => {
  const sourceClone = Array.from(source);
  const destClone = Array.from(destination);
  const [removed]: any = sourceClone.splice(droppableSource.index, 1);

  const ind = removed.lastIndexOf(sourceString);

  const [ret] = [removed.substring(0, ind) + destinationString];

  destClone.splice(droppableDestination.index, 0, ret);

  const result = {};
  result[droppableSource.droppableId] = sourceClone;
  result[droppableDestination.droppableId] = destClone;

  return result;
};

export default function KanBan() {
  const [items, setItems] = useState<Map<string, string[]>>(new Map());
  const [open, setOpen] = useState(false);
  const [colName, setColName] = useState('');

  const [selectedEntry, setSelectedEntry] = useState(-1);
  const [showSelectedEntry, setShowSelectedEntry] = useState(false);

  const classes = useStyles();
  const dispatch = useDispatch();
  const columns = useSelector(getColumnNames);
  const entries = useSelector(getEntries);

  function addE(name: string) {
    const nextId = entries.length + 1;
    const toAdd = `${nextId},tt,tt,tt,tt,Dec 30 2020,${name}`;
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    const tmp: string[] = items.get(name)!;
    setItems(new Map(items.set(name, tmp.concat([toAdd]))));
    dispatch(addEntry(toAdd));
  }

  function addC(name: string) {
    if (!items.has(name)) {
      setItems(new Map(items.set(name, [])));
      dispatch(addColumn(name));
    }
  }

  function removeC(name: string) {
    items.delete(name);
    setItems(new Map(items));
    dispatch(removeColumn(name));
  }

  function openCardInfo(info: string) {
    setSelectedEntry(entries.indexOf(info));
    setShowSelectedEntry(true);
  }

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setColName('');
    setOpen(false);
  };

  const handleCreate = () => {
    addC(colName);
    handleClose();
  };

  const handleEntryClose = () => {
    setSelectedEntry(-1);
    setShowSelectedEntry(false);
  };

  const handleEntrySave = (index: number, newEntry: string) => {
    dispatch(replaceEntry([index, newEntry]));
    handleEntryClose();
  };

  useEffect(() => {
    columns.forEach((col) => {
      setItems(
        new Map(
          items.set(
            col,
            entries.filter((e) => e.split(',')[6] === col)
          )
        )
      );
    });
  }, [entries]);

  function onDragEnd(result: DragProps) {
    const { type, source, destination } = result;

    if (!destination) return;

    if (type === 'ENTRY') {
      const sInd: any = source.droppableId;
      const dInd: any = destination.droppableId;

      if (sInd === dInd) {
        const i: any = reorder(
          items.get(sInd),
          source.index,
          destination.index
        );
        setItems(new Map(items.set(sInd, i)));
      } else {
        const res: Record<string, string[]> = move(
          sInd,
          dInd,
          items.get(sInd),
          items.get(dInd),
          source,
          destination
        );

        const tmp = items;
        tmp.set(sInd, res[sInd]);
        tmp.set(dInd, res[dInd]);

        setItems(new Map(tmp));
      }
      dispatch(setEntries([Array.from(items.values()).flat(), true]));
    } else if (type === 'COLS') {
      const s: any = reorder(columns, source.index, destination.index);

      dispatch(setColumns([s, true]));
    }
  }

  return (
    <div className={classes.page}>
      <Button
        variant="contained"
        color="primary"
        style={{
          margin: '10px',
          maxWidth: 'max-content',
          display: 'flex',
          flexDirection: 'row',
          justifyContent: 'flex-start',
        }}
        onClick={handleClickOpen}
      >
        <AddCircleIcon style={{ marginRight: '10px' }} />
        Add another column
      </Button>
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby="form-dialog-title"
      >
        <DialogContent>
          <DialogContentText>Add Another Column</DialogContentText>
          <TextField
            autoFocus
            margin="dense"
            id="name"
            label="Column Name"
            fullWidth
            value={colName}
            onChange={(e) => setColName(e.target.value)}
          />
        </DialogContent>
        <DialogActions>
          <Button onClick={handleCreate} color="primary">
            Create
          </Button>
          <Button onClick={handleClose} color="primary">
            Cancel
          </Button>
        </DialogActions>
      </Dialog>
      <DragDropContext onDragEnd={onDragEnd}>
        <Droppable droppableId="board" type="COLS" direction="horizontal">
          {(provided) => (
            <div
              className={classes.root}
              ref={provided.innerRef}
              {...provided.droppableProps}
            >
              {columns.map((c, index) => (
                <ColumnCard
                  key={c}
                  name={c}
                  entries={items.get(c)}
                  addE={addE}
                  removeC={removeC}
                  index={index}
                  openCardInfo={openCardInfo}
                />
              ))}
              {provided.placeholder}
            </div>
          )}
        </Droppable>
      </DragDropContext>
      <EditEntry
        index={selectedEntry}
        entry={selectedEntry !== -1 ? entries[selectedEntry] : ''}
        handleEntryClose={handleEntryClose}
        showSelectedEntry={showSelectedEntry}
        handleEntrySave={handleEntrySave}
      />
    </div>
  );
}
