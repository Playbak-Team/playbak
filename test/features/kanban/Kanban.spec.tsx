import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import Kanban from '../../../app/features/kanban/Kanban';
import * as kanbanSlice from '../../../app/features/kanban/kanbanSlice';
import { KanbanStateInterface } from '../../../app/interfaces';

Enzyme.configure({ adapter: new Adapter() });

function setup(
  preloadedState: {
    kanban: KanbanStateInterface;
  } = {
    kanban: {
      columns: ['todo', 'inprogress', 'done'],
      entries: [
        '1,title1,subtitle1,desc1,label1,Dec 30 2020,todo',
        '2,title2,subtitle2,desc2,label2,Dec 30 2020,todo',
        '3,title3,subtitle3,desc3,label3,Dec 30 2020,todo',
      ],
    },
  }
) {
  const store = configureStore({
    reducer: { kanban: kanbanSlice.default },
    preloadedState,
  });

  const getWrapper = () =>
    mount(
      <Provider store={store}>
        <Router>
          <Kanban />
        </Router>
      </Provider>
    );
  const component = getWrapper();
  return {
    store,
    component,
  };
}

describe('Profile component', () => {
  it('should match exact snapshot', () => {
    const { store } = setup();
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Kanban />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
