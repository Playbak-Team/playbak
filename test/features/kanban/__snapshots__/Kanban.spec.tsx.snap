// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component should match exact snapshot 1`] = `
<div
  className="makeStyles-page-2"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": "10px",
        "maxWidth": "max-content",
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "marginRight": "10px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
        />
      </svg>
      Add another column
    </span>
  </button>
  <div
    className="makeStyles-root-1"
    data-rbd-droppable-context-id="1"
    data-rbd-droppable-id="board"
  >
    <div
      aria-describedby="rbd-hidden-text-1-hidden-text-11"
      className="makeStyles-col-3"
      data-rbd-drag-handle-context-id="1"
      data-rbd-drag-handle-draggable-id="todo"
      data-rbd-draggable-context-id="1"
      data-rbd-draggable-id="todo"
      draggable={false}
      onDragStart={[Function]}
      onTransitionEnd={null}
      role="button"
      style={
        Object {
          "backgroundColor": "#FFCF99",
          "transform": null,
          "transition": null,
          "userSelect": "none",
        }
      }
      tabIndex={0}
    >
      <div
        className="makeStyles-coltitle-4"
      >
        <h3>
          todo
        </h3>
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="makeStyles-containerparent-5"
      >
        <div
          className="makeStyles-container-6"
          data-rbd-droppable-context-id="1"
          data-rbd-droppable-id="todo"
          style={
            Object {
              "background": "rgb(235,236,240)",
            }
          }
        >
          <div
            className="makeStyles-entriescontainer-7"
          >
            <div />
          </div>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "margin": "10px",
            "minWidth": "90%",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "marginRight": "5px",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          Add another card
        </span>
      </button>
    </div>
    <div
      aria-describedby="rbd-hidden-text-1-hidden-text-11"
      className="makeStyles-col-3"
      data-rbd-drag-handle-context-id="1"
      data-rbd-drag-handle-draggable-id="inprogress"
      data-rbd-draggable-context-id="1"
      data-rbd-draggable-id="inprogress"
      draggable={false}
      onDragStart={[Function]}
      onTransitionEnd={null}
      role="button"
      style={
        Object {
          "backgroundColor": "#FFCF99",
          "transform": null,
          "transition": null,
          "userSelect": "none",
        }
      }
      tabIndex={0}
    >
      <div
        className="makeStyles-coltitle-4"
      >
        <h3>
          inprogress
        </h3>
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="makeStyles-containerparent-5"
      >
        <div
          className="makeStyles-container-6"
          data-rbd-droppable-context-id="1"
          data-rbd-droppable-id="inprogress"
          style={
            Object {
              "background": "rgb(235,236,240)",
            }
          }
        >
          <div
            className="makeStyles-entriescontainer-7"
          >
            <div />
          </div>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "margin": "10px",
            "minWidth": "90%",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "marginRight": "5px",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          Add another card
        </span>
      </button>
    </div>
    <div
      aria-describedby="rbd-hidden-text-1-hidden-text-11"
      className="makeStyles-col-3"
      data-rbd-drag-handle-context-id="1"
      data-rbd-drag-handle-draggable-id="done"
      data-rbd-draggable-context-id="1"
      data-rbd-draggable-id="done"
      draggable={false}
      onDragStart={[Function]}
      onTransitionEnd={null}
      role="button"
      style={
        Object {
          "backgroundColor": "#FFCF99",
          "transform": null,
          "transition": null,
          "userSelect": "none",
        }
      }
      tabIndex={0}
    >
      <div
        className="makeStyles-coltitle-4"
      >
        <h3>
          done
        </h3>
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="makeStyles-containerparent-5"
      >
        <div
          className="makeStyles-container-6"
          data-rbd-droppable-context-id="1"
          data-rbd-droppable-id="done"
          style={
            Object {
              "background": "rgb(235,236,240)",
            }
          }
        >
          <div
            className="makeStyles-entriescontainer-7"
          >
            <div />
          </div>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "margin": "10px",
            "minWidth": "90%",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "marginRight": "5px",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          Add another card
        </span>
      </button>
    </div>
  </div>
</div>
`;
